qiskit/compiler/transpile.py

```diff
@@ -205,10 +229,16 @@ def callback_func(**kwargs):
         config = user_config.get_config()
         optimization_level = config.get('transpile_optimization_level', 1)
 
+    if scheduling_method is not None and backend is None and not instruction_durations:
+        warnings.warn("When scheduling circuits without backend,"
+                      " 'instruction_durations' should be usually provided.",
+                      UserWarning)
+
```

Comment:
```suggestion
        warnings.warn("When scheduling circuits without backend,"
                      " 'instruction_durations' should usually be provided.",
                      UserWarning)

```
How does it schedule without duration information?

Reply:
Good question. Some instructions, e.g. Delays and Barriers, can be scheduled without backend or 'instruction_durations'. So I think circuits composed of only such instructions should be schedulable (=> Don't raise error here). I don't know when such circuits are useful, though (=> Warn here instead).